### PR TITLE
[CHORE] Support empty title & alias on the message

### LINKED ISSUE #2323: [NEW][RCVC] Add a placeholder title to attachments with `fields` but no title
Add the default title: "No title"

### PR DESCRIPTION
**Pull request by rafaelks**
@RocketChat/ios

Closes #2323 

simulator screen shot - iphone se - 2018-11-22 at 00 16 50
simulator screen shot - iphone se - 2018-11-22 at 00 16 48

### CODE REVIEW

File: Rocket.Chat/Views/Chat/New Chat/ChatItems/AudioMessageChatItem.swift [hunk 1]
```diff
@@ -23,7 +23,15 @@ class AudioMessageChatItem: BaseMessageChatItem, ChatItem, Differentiable {
...
         self.audioURL = audioURL
         self.hasText = hasText
-        super.init(user: user, avatar: message?.avatar, emoji: message?.emoji, date: message?.createdAt, isUnread: message?.unread ?? false)
+
+        super.init(
+            user: user,
+            avatar: message?.avatar,
+            emoji: message?.emoji,
+            alias: message?.alias,
+            date: message?.createdAt,
```
Comment (cardoso) [inline Rocket.Chat/Views/Chat/New Chat/ChatItems/AudioMessageChatItem.swift]: 👍

File: Rocket.Chat/Views/Chat/New Chat/ChatItems/BaseMessageChatItem.swift [hunk 2]
```diff
@@ -22,10 +23,11 @@ class BaseMessageChatItem {
         return RCDateFormatter.time(date)
     }
 
-    init(user: UnmanagedUser?, avatar: String?, emoji: String?, date: Date?, isUnread: Bool = false) {
+    init(user: UnmanagedUser?, avatar: String?, emoji: String?, alias: String?, date: Date?, isUnread: Bool = false) {
```
Comment (cardoso) [inline Rocket.Chat/Views/Chat/New Chat/ChatItems/BaseMessageChatItem.swift]: What about doing the same thing here as you did in the other places?

File: Rocket.Chat/Views/Chat/New Chat/ChatItems/BaseTextAttachmentChatItem.swift [hunk 3]
```diff
@@ -11,8 +11,16 @@ import Foundation
 class BaseTextAttachmentChatItem: BaseMessageChatItem {
     let collapsed: Bool
 
-    init(collapsed: Bool, user: UnmanagedUser?, avatar: String?, emoji: String?, date: Date?, isUnread: Bool = false) {
+    init(collapsed: Bool, user: UnmanagedUser?, avatar: String?, emoji: String?, alias: String?, date: Date?, isUnread: Bool = false) {
```
[the remark this answers is not in the document]
Comment (cardoso) [reply Rocket.Chat/Views/Chat/New Chat/ChatItems/BaseTextAttachmentChatItem.swift]: Also the same thing here to remove this warning

File: Rocket.Chat/Views/Chat/New Chat/ChatItems/ImageMessageChatItem.swift [hunk 4]
```diff
@@ -21,13 +21,30 @@ final class ImageMessageChatItem: BaseMessageChatItem, ChatItem, Differentiable
     var imageURL: URL?
     let hasText: Bool
 
-    init(identifier: String, title: String?, descriptionText: String?, imageURL: URL?, hasText: Bool, user: UnmanagedUser?, message: UnmanagedMessage?) {
+    init(
+        identifier: String,
+        title: String?,
+        descriptionText: String?,
```
Comment (cardoso) [inline Rocket.Chat/Views/Chat/New Chat/ChatItems/ImageMessageChatItem.swift]: 👍

File: Rocket.Chat/Views/Chat/New Chat/ChatItems/SequentialMessageChatItem.swift [hunk 5]
```diff
@@ -19,7 +19,15 @@ final class SequentialMessageChatItem: BaseMessageChatItem, ChatItem, Differenti
 
     init(user: UnmanagedUser, message: UnmanagedMessage) {
         self.message = message
-        super.init(user: user, avatar: message.avatar, emoji: message.emoji, date: message.createdAt, isUnread: message.unread)
+
+        super.init(
+            user: user,
+            avatar: message.avatar,
+            emoji: message.emoji,
```
Comment (cardoso) [inline Rocket.Chat/Views/Chat/New Chat/ChatItems/SequentialMessageChatItem.swift]: 👍

File: Rocket.Chat/Views/Chat/New Chat/ChatItems/ImageMessageChatItem.swift [hunk 6]
```diff
@@ -21,13 +21,30 @@ final class ImageMessageChatItem: BaseMessageChatItem, ChatItem, Differentiable
...
         self.identifier = identifier
         self.title = title
         self.descriptionText = descriptionText
         self.imageURL = imageURL
         self.hasText = hasText
-        super.init(user: user, avatar: message?.avatar, emoji: message?.emoji, date: message?.createdAt, isUnread: message?.unread ?? false)
+
+        super.init(
+            user: user,
+            avatar: message?.avatar,
```
Comment (cardoso) [inline Rocket.Chat/Views/Chat/New Chat/ChatItems/ImageMessageChatItem.swift]: 👍

File: Rocket.Chat/Views/Chat/New Chat/ChatItems/TextAttachmentChatItem.swift [hunk 7]
```diff
@@ -23,14 +23,34 @@ final class TextAttachmentChatItem: BaseTextAttachmentChatItem, ChatItem, Differ
     let color: String?
     let hasText: Bool
 
-    init(identifier: String, fields: [UnmanagedField], title: String, subtitle: String?, color: String?, collapsed: Bool, hasText: Bool, user: UnmanagedUser?, message: UnmanagedMessage?) {
+    init(
+        identifier: String,
+        fields: [UnmanagedField],
+        title: String,
```
Comment (cardoso) [inline Rocket.Chat/Views/Chat/New Chat/ChatItems/TextAttachmentChatItem.swift]: 👍

File: Rocket.Chat/Views/Chat/New Chat/ChatItems/TextAttachmentChatItem.swift [hunk 8]
```diff
@@ -23,14 +23,34 @@ final class TextAttachmentChatItem: BaseTextAttachmentChatItem, ChatItem, Differ
...
         self.subtitle = subtitle
         self.color = color
         self.fields = fields
         self.hasText = hasText
-        super.init(collapsed: collapsed, user: user, avatar: message?.avatar, emoji: message?.emoji, date: message?.createdAt, isUnread: message?.unread ?? false)
+
+        super.init(
+            collapsed: collapsed,
+            user: user,
+            avatar: message?.avatar,
```
Comment (cardoso) [inline Rocket.Chat/Views/Chat/New Chat/ChatItems/TextAttachmentChatItem.swift]: 👍

File: Rocket.Chat/Views/Chat/New Chat/ChatItems/VideoMessageChatItem.swift [hunk 9]
```diff
@@ -21,13 +21,29 @@ final class VideoMessageChatItem: BaseMessageChatItem, ChatItem, Differentiable
     let videoThumbPath: URL?
     let hasText: Bool
 
-    init(identifier: String, descriptionText: String?, videoURL: URL?, videoThumbPath: URL?, hasText: Bool, user: UnmanagedUser?, message: UnmanagedMessage?) {
+    init(
+        identifier: String,
+        descriptionText: String?,
```
Comment (cardoso) [inline Rocket.Chat/Views/Chat/New Chat/ChatItems/VideoMessageChatItem.swift]: 👍

File: Rocket.Chat/Views/Chat/New Chat/ChatItems/VideoMessageChatItem.swift [hunk 10]
```diff
@@ -21,13 +21,29 @@ final class VideoMessageChatItem: BaseMessageChatItem, ChatItem, Differentiable
...
         self.identifier = identifier
         self.descriptionText = descriptionText
         self.videoURL = videoURL
         self.videoThumbPath = videoThumbPath
         self.hasText = hasText
-        super.init(user: user, avatar: message?.avatar, emoji: message?.emoji, date: message?.createdAt)
+
+        super.init(
+            user: user,
+            avatar: message?.avatar,
```
Comment (cardoso) [inline Rocket.Chat/Views/Chat/New Chat/ChatItems/VideoMessageChatItem.swift]: 👍

File: Rocket.Chat/Views/Chat/New Chat/ChatItems/BaseMessageChatItem.swift (hunk 2 above)
Comment (rafaelks) [reply Rocket.Chat/Views/Chat/New Chat/ChatItems/BaseMessageChatItem.swift]: Yes! :-)

**Pull request merged**